src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -513,3 +525,57 @@ def _convert_to_stopping_conditions_json(
                 threshold.to_json for threshold in model_latency_thresholds
             ]
         return stopping_conditions
+
+    def _get_right_size_and_model_recommendation(
+        self,
+        model_res=None,
+        recommendation_res=None,
+        inference_recommendation_id=None,
+    ):
+        """Get recommendation from right size job or model"""
+        right_size_recommendation, model_recommendation = None, None
+        if recommendation_res:
+            right_size_recommendation = self._get_recommendation(
+                recommendation_list=recommendation_res["InferenceRecommendations"],
+                inference_recommendation_id=inference_recommendation_id,
+            )
+        if model_res:
+            model_recommendation = self._get_recommendation(
+                recommendation_list=model_res["DeploymentRecommendation"][
+                    "RealTimeInferenceRecommendations"
+                ],
+                inference_recommendation_id=inference_recommendation_id,
+            )
+        if right_size_recommendation is None and model_recommendation is None:
+            raise ValueError("Inference Recommendation id is not valid")
+
+        return right_size_recommendation, model_recommendation
+
+    def _get_recommendation(self, recommendation_list, inference_recommendation_id):
+        """Get recommendation based on recommendation id"""
+        return next(
+            (
+                rec
+                for rec in recommendation_list
+                if rec["RecommendationId"] == inference_recommendation_id
+            ),
+            None,
+        )
+
+    def _describe_recommendation_job_and_model(self, sage_client, job_or_model_name):
+        """Describe inference recommendation job and model results"""
+        recommendation_res, model_res = None, None
+        try:
+            recommendation_res = sage_client.describe_inference_recommendations_job(
+                JobName=job_or_model_name
+            )
+        except sage_client.exceptions.ResourceNotFound:
+            pass
+        try:
+            model_res = sage_client.describe_model(ModelName=job_or_model_name)
```

Comment:
we should first check the recommendations job results. If the recommendation id is there, then do not call describe_model because it is a wasted API call.